lib/puppet/file_serving/metadata.rb

```diff
@@ -31,14 +31,14 @@ def source_permissions=(source_permissions)
 
   def content_uri=(path)
     begin
-      uri = URI.parse(URI.escape(path))
+      uri = URI.parse(Puppet::Util.uri_escape(path))
     rescue URI::InvalidURIError => detail
       raise(ArgumentError, "Could not understand URI #{path}: #{detail}")
     end
     raise(ArgumentError, "Cannot use opaque URLs '#{path}'") unless uri.hierarchical?
     raise(ArgumentError, "Must use URLs of type puppet as content URI") if uri.scheme != "puppet"
 
-    @content_uri = path
+    @content_uri = path.encode(Encoding::UTF_8)
```

Comment:
Nope, not anymore - `uri_escape` will throw when `path` is `nil` (given I fixed it per your suggestion to match previous behavior)